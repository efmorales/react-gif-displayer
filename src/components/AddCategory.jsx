import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length <= 1) return;

        // setCategories( categories => [inputValue, ...categories]);
        onNewCategory(inputValue.trim());
        setInputValue('');

    }

    return (
        <form onSubmit={ (e) => onSubmit(e)}>

            <input
                type="text"
                placeholder="Search gifs"
                value={inputValue}
                onChange={onInputChange}
            />

        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}