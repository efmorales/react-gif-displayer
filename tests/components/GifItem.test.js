import { GifItem } from "../../src/components/GifItem";
import { screen, render, fireEvent } from "@testing-library/react";


describe('tests for GifItem component', () => { 
    
    test('should match with the snapshot', () => { 

        const {container} = render(<GifItem id="1" url="https://localhost/image.jpg" title="A title" />)
        
        expect(container).toMatchSnapshot()



     })
 })