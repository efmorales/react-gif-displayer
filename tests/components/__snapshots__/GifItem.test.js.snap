// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for GifItem component should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="A title"
      src="https://localhost/image.jpg"
    />
    <p>
      A title
    </p>
  </div>
</div>
`;
